src/components/tree/phyloTree/helpers.js

```diff
@@ -128,32 +128,63 @@ function isWithinBranchTolerance(node, otherNode, distanceMeasure) {
     return (getTraitFromNode(node, "num_date") - tolerance < getTraitFromNode(otherNode, "num_date"));
   }
-  /* Compute the divergence tolerance similarly to above. This uses the approach used to compute the
-  x-axis grid within phyotree, and could be refactored into a helper function. Note that we don't store
-  the maximum divergence on the tree so we use the in-view max instead */
+
+  /* If mutations aren't defined, we fallback to calculating divergence tolerance similarly to temporal.
+  This uses the approach used to compute the x-axis grid within phyotree, and could be refactored into a
+  helper function. Note that we don't store the maximum divergence on the tree so we use the in-view max instead */
   const tolerance = (node.shell.that.xScale.domain()[1] - node.shell.that.nodes[0].depth)/50;
   return (getDivFromNode(node) - tolerance < getDivFromNode(otherNode));
 }
 
+/**
+ * Walk up the tree from node until we find either a node which has a nucleotide mutation or we
+ * reach the root of the (sub)tree. Gaps, deletions and undeletions do not count as mutations here.
+ */
+function findFirstBranchWithAMutation(node) {
+  if (node.parent === node) {
+    return node;
+  }
+  const categorisedMutations = getBranchMutations(node, {}); // 2nd param of `{}` means we skip homoplasy detection
+  if (categorisedMutations?.nuc?.unique?.length>0) {
+    return node.parent;
+  }
+
+  return findFirstBranchWithAMutation(node.parent);
+}
 
 /**
  * Given a `node`, get the parent, grandparent etc node which is beyond some
  * branch length threshold (either divergence or time). This is useful for finding the node
- * beyond a polytomy, or polytomy-like structure
+ * beyond a polytomy, or polytomy-like structure. If nucleotide mutations are defined on
+ * the tree (and distanceMeasure=div) then we find the first branch with a mutation.
  * @param {object} node - tree node
- * @param {string} getParentBeyondPolytomy -- 'num_date' or 'div'
+ * @param {string} distanceMeasure -- 'num_date' or 'div'
+ * @param {object} observedMutations
  * @returns {object} the closest node up the tree (towards the root) which is beyond
  * some threshold
  */
-export const getParentBeyondPolytomy = (node, distanceMeasure) => {
+export const getParentBeyondPolytomy = (node, distanceMeasure, observedMutations) => {
   let potentialNode = node.parent;
+  if (distanceMeasure==="div" && areNucleotideMutationsPresent(observedMutations)) {
+    return findFirstBranchWithAMutation(node);
+  }
   while (isWithinBranchTolerance(node, potentialNode, distanceMeasure)) {
     if (potentialNode === potentialNode.parent) break; // root node of tree
     potentialNode = potentialNode.parent;
   }
   return potentialNode;
 };
 
+function areNucleotideMutationsPresent(observedMutations) {
+  const mutList = Object.keys(observedMutations);
+  for (let idx=mutList.length-1; idx>=0; idx--) { // start at end, as nucs come last in the key-insertion order
```

Comment:
Can you point to where the key-insertion order can be found? (I'm still trying to piece everything together 😅 )

From what I can find, it looks like the order is dependent on keys in `n.branch_attrs.mutations`

https://github.com/nextstrain/auspice/blob/07b66e80c61aea344b73f8c9aee12213a54c1a60/src/util/treeJsonProcessing.js#L145-L156

Which I think is based on the order of insertion [within `augur export`](https://github.com/nextstrain/augur/blob/e1e8bb155c25a2901bcd41423b0f67d14386b6b3/augur/export_v2.py#L700-L707)? 
(boo, cross-repo code snippets don't work)
```python
    def _transfer_mutations(node, raw_data):
        if "aa_muts" in raw_data or "muts" in raw_data:
            node["branch_attrs"]["mutations"] = {}
            if "muts" in raw_data and len(raw_data["muts"]):
                node["branch_attrs"]["mutations"]["nuc"] = raw_data["muts"]
            if "aa_muts" in raw_data:
                aa = {gene:data for gene, data in raw_data["aa_muts"].items() if len(data)}
                node["branch_attrs"]["mutations"].update(aa)
```

Reply:
I don't actually know - but I would presume you're right here. This means it's related to https://github.com/nextstrain/augur/issues/1052, which if we move to OrderedDict will make "nuc" come first for GenBank files (created by `load_features`, since "source" comes before CDSs in genbank files) and last for VCF files (as the translate code adds them explicitly). (That's from reading the code, I haven't double checked everything.)

In testing for this PR the "nuc" block consistently came last. In terms of code performance it's not overly important, but it seemed sensible to iterate this way. I'll make a note on the augur issue to revisit this auspice code in the future.

Reply:
A further note: it would be more efficient to calculate and store a "areNucleotideMutationsPresent" flag in a redux store rather than re-calculating it each time one presses the "goBack" button. I decided that the slight performance hit (not noticeable in my testing) was preferable to the overhead of this state & keeping it in sync as trees change etc.